plugins/GSdx/GSUtil.cpp

```diff
@@ -396,6 +396,39 @@ D3D_FEATURE_LEVEL GSUtil::CheckDirect3D11Level(IDXGIAdapter *adapter, D3D_DRIVER
 	return SUCCEEDED(hr) ? level : (D3D_FEATURE_LEVEL)0;
 }
 
+GSRendererType GSUtil::GetBestRenderer()
+{
+	CComPtr<IDXGIAdapter1> adapter;
+	DXGI_ADAPTER_DESC1 desc;
+	CComPtr<IDXGIFactory1> dxgi_factory;
+	GSRendererType BestRenderer = GSRendererType::Default;
+
+	if (!GSUtil::CheckDXGI() || !GSUtil::CheckD3D11())
+		return GSRendererType::DX9_HW;
+
+	if (SUCCEEDED(CreateDXGIFactory1(IID_PPV_ARGS(&dxgi_factory))))
+	{
+		if (SUCCEEDED(dxgi_factory->EnumAdapters1(NULL, &adapter)))
+		{
+			D3D_FEATURE_LEVEL level = GSUtil::CheckDirect3D11Level();
+
+			if (SUCCEEDED(adapter->GetDesc1(&desc)))
```

Comment:
> If anything up to this stage fails you return GSRendererType::Default, which is GSRendererType::Undefined. Unintended?

That was intended indeed but I guess it'd be better to just return ``GSRendererType::DX9_HW`` at such cases.

About other remarks : They have been fixed.